packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -0,0 +1,222 @@
+import { GrafanaTheme } from '@grafana/data';
+import { useTheme, styleMixins } from '../../themes';
+import { Tooltip, PopoverContent } from '../Tooltip/Tooltip';
+import { OnTagClick } from '../Tags/Tag';
+import { TagList } from '../Tags/TagList';
+
+interface ContainerProps extends HTMLAttributes<HTMLOrSVGElement> {
+  /** Customise the container html element for the card. Defaults to div */
+  tag?: keyof JSX.IntrinsicElements;
+  /** Content for the card's tooltip */
+  tooltip?: PopoverContent;
+}
+
+const CardContainer: FC<ContainerProps> = ({ tag = 'div', children, tooltip, ...props }) => {
+  const element = React.createElement(tag, props, children);
+  return tooltip ? (
+    <Tooltip placement="top" content={tooltip} theme="info">
+      {element}
+    </Tooltip>
+  ) : (
+    element
+  );
+};
+
+export interface CardInnerProps {
+  href?: string;
+}
+
+const CardInner: FC<CardInnerProps> = ({ children, href }) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+  return href ? (
+    <a className={styles.innerLink} href={href}>
+      {children}
+    </a>
+  ) : (
+    <>{children}</>
+  );
+};
+
+export interface Props extends ContainerProps {
+  /** Main heading for the Card **/
+  heading: ReactNode;
+  /** Additional data about the card. If array is supplied, elements will be rendered with vertical line separator */
+  metaData?: ReactNode | ReactNode[];
+  /** Card description text */
+  description?: string;
+  /** List of tags to display in the card */
+  tags?: string[];
+  /** Optional callback for tag onclick event */
+  onTagClick?: OnTagClick;
+  /** Indicates if the card and all its actions can be interacted with */
+  disabled?: boolean;
+  /** Image or icon to be displayed on the let side of the card */
+  mediaContent?: ReactNode;
+  /** Main card actions **/
+  actions?: ReactElement[];
+  /** Right-side actions */
+  secondaryActions?: ReactElement[];
+  /** Link to redirect to on card click. If provided, the Card inner content will be rendered inside `a` */
+  href?: string;
+  /** On click handler for the Card */
+  onClick?: () => void;
+}
+
+export const Card: FC<Props> = ({
+  heading,
+  description,
+  metaData,
+  tags = [],
+  onTagClick,
+  disabled,
+  mediaContent,
+  actions = [],
+  tooltip,
+  secondaryActions = [],
+  tag,
+  href,
+  onClick,
+  className,
+  ...htmlProps
+}) => {
+  const hasActions = Boolean(actions.length || secondaryActions.length);
+  const disableHover = disabled || actions.length > 1;
+  const theme = useTheme();
+  const styles = getStyles(theme, disabled && !actions.length, disableHover);
+  // Join meta data elements by '|'
+  const meta = Array.isArray(metaData)
+    ? (metaData as ReactNode[]).filter(Boolean).reduce((prev, curr, i) => [
+        prev,
+        <span key={`separator_${i}`} className={styles.separator}>
+          |
+        </span>,
+        curr,
+      ])
+    : metaData;
+  const onCardClick = disableHover ? () => {} : onClick;
```

Comment:
```suggestion
  const onCardClick = useCallback((e) => {
    if(disableHover) {
      return null;
    }
    
    onClick(e)
  } , [onClick, disableHover]);
```

or sth similar. Otherwise, you are creating noop with every render.